test/miner_blockchain_SUITE.erl

```diff
@@ -334,14 +338,160 @@ election_test(Config) ->
     Stop2 = miner_ct_utils:stop_miners(StopList),
 
     %% sleep a lil then start the nodes back up again
-    timer:sleep(2000),
+    timer:sleep(1000),
 
     miner_ct_utils:start_miners(Stop2),
 
     %% fourth: confirm that blocks and elections are proceeding
-    ok = miner_ct_utils:wait_for_gte(epoch, Miners, ElectionEpoch + 1),
-    ok.
+    ok = miner_ct_utils:wait_for_gte(epoch, Miners, ElectionEpoch + 1).
+
+election_multi_test(Config) ->
+    BaseDir = ?config(base_dir, Config),
+    %% get all the miners
+    Miners = ?config(miners, Config),
+    %% AddrList = ?config(tagged_miner_addresses, Config),
+    Addresses = ?config(addresses, Config),
+    {Priv, _Pub} = ?config(master_key, Config),
+
+    %% we wait until we have seen all miners hit an epoch of 2
+    ok = miner_ct_utils:wait_for_gte(epoch, Miners, 2, all, 90),
+
+    ct:pal("starting multisig attempt"),
+
+    #{secret := Priv1, public := Pub1} = libp2p_crypto:generate_keys(ecc_compact),
+    #{secret := Priv2, public := Pub2} = libp2p_crypto:generate_keys(ecc_compact),
+    #{secret := Priv3, public := Pub3} = libp2p_crypto:generate_keys(ecc_compact),
+    #{secret := Priv4, public := Pub4} = libp2p_crypto:generate_keys(ecc_compact),
+    #{secret := Priv5, public := Pub5} = libp2p_crypto:generate_keys(ecc_compact),
+    BinPub1 = libp2p_crypto:pubkey_to_bin(Pub1),
+    BinPub2 = libp2p_crypto:pubkey_to_bin(Pub2),
+    BinPub3 = libp2p_crypto:pubkey_to_bin(Pub3),
+    BinPub4 = libp2p_crypto:pubkey_to_bin(Pub4),
+    BinPub5 = libp2p_crypto:pubkey_to_bin(Pub5),
+
+    Txn6 = vars(#{?use_multi_keys => true}, 2, Priv),
+    _ = [ok = ct_rpc:call(M, blockchain_worker, submit_txn, [Txn6]) || M <- Miners],
+
+    ok = miner_ct_utils:wait_for_chain_var_update(Miners, ?use_multi_keys, true),
+
+    Txn7_0 = blockchain_txn_vars_v1:new(
+               #{}, 3,
+               #{multi_keys => [BinPub1, BinPub2, BinPub3, BinPub4, BinPub5]}),
+    Proofs7 = [blockchain_txn_vars_v1:create_proof(P, Txn7_0)
+               || P <- [Priv1, Priv2, Priv3, Priv4, Priv5]],
+    Txn7_1 = blockchain_txn_vars_v1:multi_key_proofs(Txn7_0, Proofs7),
+    Proof7 = blockchain_txn_vars_v1:create_proof(Priv, Txn7_1),
+    Txn7 = blockchain_txn_vars_v1:proof(Txn7_1, Proof7),
+    _ = [ok = ct_rpc:call(M, blockchain_worker, submit_txn, [Txn7]) || M <- Miners],
+
+    ct:pal("transitioned to multikey"),
```

Comment:
Should we use wait_for_chain_var_update here too?

Reply:
yeah, fixing